compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -339,6 +339,10 @@ fn mir_promoted(
         &[
             &promote_pass,
             &simplify::SimplifyCfg::new("promote-consts"),
+            // These next passes must be executed together
+            &add_call_guards::CriticalCallEdges,
```

Comment:
Doing `CriticalCallEdges` before optimizations means that all optimizations now have to maintain the invariant that the destination edge of a call is not a critical edge. `SimplifyCfg` doesn't maintain this invariant afaik. Violating this invariant will result in codegen backends incorrectly handing return value storage for functions that can unwind.